src/templates/actionBar.jsx

```diff
@@ -136,185 +164,217 @@ class ActionBar extends React.PureComponent {
-                                                                id={iconTable.id}
-                                                                onClick={iconTable.onClick}
-                                                                title={iconTable.title || 'Table View'}
-                                                                type="list"
-                                                            />
-                                                            )}
+                                                    })}
+                                                </DropdownButton>
+                                            )}
 
-                                                            {jsx}
-                                                        </List.Item>
-                                                    );
-                                                })}
-                                            </List>
+                                            {list && _.isArray(list) && (
+                                                <List
+                                                    className="action_bar--list"
+                                                    horizontal
+                                                >
+                                                    {_.map(list, (item) => {
+                                                        const {
+                                                            actionsButton,
+                                                            jsx, // eslint-disable-line no-shadow
+                                                            iconBack,
+                                                            iconFilter,
+                                                            iconGrid,
+                                                            iconSearch,
+                                                            iconTable,
+                                                        } = item;
+
+                                                        const divide = _.isUndefined(item.divide) ||
+                                                            item.divide;
+
+                                                        const listItemClassName = ClassNames('action_bar--list_item', {
+                                                            'action_bar--list_item-jsx': jsx,
+                                                            'action_bar--list_item-icon_filter': iconFilter,
+                                                            'action_bar--list_item-icon_grid': iconGrid,
+                                                            'action_bar--list_item-icon_table': iconTable,
+                                                        });
+
+                                                        const itemKey = `action_bar--list_item-${gridColumnListItemKeyNum++}`; // eslint-disable-line no-plusplus
```

Comment:
Haha there it is! 😁 `++` FTW!

Reply:
`++` has been in so many programming languages using syntax for so long, I'm not sure when it officially because "considered harmful".  I think there are situations where you should probably avoid it (i.e. incrementing a variable and reading its value at the same time is just not smart there) but I think banning it wholesale is a bit harsh and unnecessary.